lib/internal/test_runner/mock/loader.js

```diff
@@ -137,7 +137,7 @@ async function load(url, context, nextLoad) {
 async function createSourceFromMock(mock, format) {
   // Create mock implementation from provided exports.
   const { exportNames, hasDefaultExport, url } = mock;
-  const useESM = format === 'module';
+  const useESM = format === 'module' || format === 'module-typescript';
```

Comment:
I'm not familiar enough with the TypeScript support code, so I'm just asking: does the code on lines 121-124 need to be updated as well?

Reply:
Yep, that's what I meant. I was asking specifically because of the `commonjs-typescript` type. But if you think it works, I'm good with resolving this conversation.